test/tests.ts

```diff
@@ -0,0 +1,179 @@
+  if (BuidlerContext.isCreated()) {
+    const ctx = BuidlerContext.getBuidlerContext();
+    const globalAsAny = global as any;
+    if (ctx.environment !== undefined) {
+      for (const key of Object.keys(ctx.environment)) {
+        globalAsAny[key] = undefined;
+      }
+      // unload config file too.
+      unloadModule(ctx.environment.config.paths.configFile);
+    }
+    BuidlerContext.deleteBuidlerContext();
+  }
+
+  // Unload all the buidler's entrypoints.
+  unloadModule("@nomiclabs/buidler");
+  unloadModule("@nomiclabs/buidler/config");
+  unloadModule("@nomiclabs/buidler/register");
+  unloadModule("@nomiclabs/buidler/internal/cli/cli");
+  unloadModule("@nomiclabs/buidler/internal/lib/buidler-lib");
+  unloadModule("@nomiclabs/buidler/internal/core/config/config-env");
+  unloadModule("@nomiclabs/buidler/internal/core/tasks/builtin-tasks");
+
+  // Unload bultin tasks.
+  Object.keys(require.cache).forEach(module => {
+    if (module.includes("buidler/builtin-tasks")) {
+      unloadModule(module);
+    }
+  });
+}
+
+function unloadModule(path: string) {
+  try {
+    delete require.cache[require.resolve(path)];
+  } catch (err) {
+    // module wasn't loaded
+  }
+}
+
+async function reset() {
+  await unloadBuidlerContext();
+  unloadModule("../src/index");
+  return require("@nomiclabs/buidler");
+}
+
+describe("Solpp plugin", async function() {
+  describe("js-config-project", async function() {
+    before("setup", async function() {
+      process.chdir(__dirname + "/js-config-project");
+    });
+
+    beforeEach("reset buidler context", async function() {
+      this.env = await reset();
+    });
+
+    it("should evaluate symbols as javascript functions", async function() {
+      const paths = await this.env.run(TASK_COMPILE_GET_SOURCE_PATHS);
+      const generatedContractA = readFileSync(paths[0]).toString();
+      assert.include(generatedContractA, "1337");
+    });
+
+    it("should compile without errors", async function() {
+      try {
+        await this.env.run("compile");
+      } catch (err) {
+        assert.isUndefined(err);
+      }
+    });
+  });
+
+  describe("json-config-project", async function() {
+    before("setup", async function() {
+      process.chdir(__dirname + "/json-config-project");
+    });
+
+    beforeEach("reset buidler context", async function() {
+      this.env = await reset();
+    });
+
+    it("should load definitions from json", async function() {
+      const paths = await this.env.run(TASK_COMPILE_GET_SOURCE_PATHS);
+      const generatedContractA = readFileSync(paths[0]).toString();
+
+      assert.include(generatedContractA, "48192.418291248");
+    });
+
+    it("should load the config properly", async function() {
+      assert.equal(this.env.config.solpp.collapseEmptyLines, false);
+      assert.equal(this.env.config.solpp.noFlatten, true);
+      assert.equal(this.env.config.solpp.tolerant, true);
+    });
+  });
+
+  describe("buidler-project", async function() {
```

Comment:
I don't get what's the difference between this and the other projects.

Reply:
I forgot to add some changes I had stashed, now the config definitions contains both expressions and functions.